tests/compiletest.rs

```diff
@@ -115,49 +116,22 @@ fn run_tests(mode: Mode, path: &str, target: &str, with_dependencies: bool) -> R
     let config = test_config(target, path, mode, with_dependencies);
 
     // Handle command-line arguments.
-    let mut after_dashdash = false;
-    let mut quiet = false;
-    let filters = std::env::args()
-        .skip(1)
-        .filter(|arg| {
-            if after_dashdash {
-                // Just propagate everything.
-                return true;
-            }
-            match &**arg {
-                "--quiet" => {
-                    quiet = true;
-                    false
-                }
-                "--" => {
-                    after_dashdash = true;
-                    false
-                }
-                s if s.starts_with('-') => {
-                    panic!("unknown compiletest flag `{s}`");
-                }
-                _ => true,
-            }
-        })
-        .collect::<Vec<_>>();
+    let args = ui_test::Args::test();
+    let quiet = args.quiet;
     eprintln!("   Compiler: {}", config.program.display());
     ui_test::run_tests_generic(
-        config,
+        vec![config],
+        std::env::var("MIRI_TEST_THREADS").map_or_else(
+            |_| std::thread::available_parallelism().unwrap(),
+            |threads| NonZeroUsize::new(threads.parse().unwrap()).unwrap(),
+        ),
```

Comment:
Please write this as a `match`, I find `map_or_else` much harder to read than a direct `match`